src/layouts/docs/utils/mdx-components.ts

```diff
@@ -0,0 +1,48 @@
+import Columns from 'components/author-primitives/vault/columns'
+import InlineTag from 'components/author-primitives/vault/inline-tag'
+import Placement from 'components/author-primitives/shared/placement-table'
+import NestedNode from 'components/author-primitives/waypoint/nested-node'
+import makeDocsAnchor from 'components/docs-anchor'
+import defaultMdxComponents from '@hashicorp/platform-docs-mdx'
+
+// Asana task to discuss MDX custom components in dev-portal:
+// https://app.asana.com/0/1100423001970639/1201648132084613/f
+
+const vaultMdxComponents = defaultMdxComponents({
+  // TODO: we should be able to remove the need for product here,
+  // TODO: using our existing useProductMeta utility:
+  // TODO: ref: https://github.com/hashicorp/web-platform-packages/tree/main/packages/product-meta#productmetaprovider
+  // TODO: The only piece of our MDX defaults using product
+  // TODO: is EnterpriseAlert, which rather than requiring a
+  // TODO: "product" prop, should useProductMeta
+  // TODO: ref: https://github.com/hashicorp/web-platform-packages/blob/6062939845d5d841d78afade8225e891cd428aa3/packages/docs-mdx/index.jsx#L42
+  product: 'vault',
+  additionalComponents: {
+    Columns,
+    Tag: InlineTag,
+    // TODO: apply makeDocsAnchor as a remark plugin instead?
+    // TODO: combined with above change, this could allow us
+    // TODO: to use a shared set of base components for all products,
+    // TODO: and only need to pass additionalComponents, on a per-route
+    // TODO: basis. (eg Columns & Tags are only needed on /vault/docs,
+    // TODO: not /vault/api-docs, and in fact are only needed on
+    // TODO: a single page - /vault/docs/plugin-portal.)
+    // TODO: Longer-term, we may be able to deprecate some MDX components...
+    // TODO: although this will admittedly be pretty difficult, since
+    // TODO: we need to support all past versions of documentation!
```

Comment:
Acknowledging that we can do something similar with `EnterpriseAlert`, it seems like we are just about to the point where we can pass in `additionalComponents` on a per-product basis without additional work.

I think the remark plugin would also require us to pass in `product` to it, which seems to kind of not solve the issue and just move the dependency on product elsewhere?

Reply:
It feels like this ☝️ would be similar to how we're handling it in `DocsPage` currently. Do you see issues with that approach?

Reply:
I could see an authoring benefit to _not_ requiring the product prefix in the MDX source. Without it, internal links still intuitively match with the source content file structure, and links across "boundaries" would still be possible and clear when looking at the source. We'd need some sort of validation to make sure that authors are adding the product prefix, so we're shifting the complexity we're handling here now to the authors, which I'm not sure is the best tradeoff. So I think my gut take is that not having the product prefix isn't the root of the issue 😅 

Expanding on brittle-ness a bit...I guess it depends on what the remark plugin is doing. If we're just visiting link nodes and changing the url itself, I agree that's not brittle. If we're using a remark plugin to swap links out with a component/different markup that makes me a bit nervous. Having looked into the future a bit with MDX v2, the AST structure has changed there so that is something we'd need to change when we get to that migration. I think I was being unfair there though, in general remark plugins are pretty effective!

If we don't move towards requiring the product prefix in source, the remark plugin would always need the product slug and the product's basePaths. If we apply the remark plugin during the content upload, we'll need to coordinate with the .io cutover as the product-prefixed URLs would NOT work with our current setup.

Reply:
It looks like `basePaths` is included in the context, so it should all be okay. https://github.com/hashicorp/dev-portal/pull/67/files#diff-0bf9bafd60b21f0b8784b6574b02bdd59a37ad5882e2c262e48c776b118a3358R4

No objections to the remark plugin if you wanna spike it out! Just thinking out loud that we'll likely need to pass a custom link component either way to wrap them with `next/link` where appropriate.

Reply:
Update here: have left the existing `DocsAnchor` `adjustedHref` logic in place, since it's already working 🎉 Did end up adding a `remark` plugin to adjust link `url` values, but for the related-but-separate issue of Vault's `/api` links needing to be corrected to `/api-docs`. Let me know if that seems like a good approach (for now at least!) 🙇